Content.Shared/Doors/Systems/PryingSystem.cs

```diff
@@ -0,0 +1,131 @@
+    }
+
+    /// <summary>
+    ///     Pry open a door.
+    /// </summary>
+    public bool TryPry(EntityUid target, EntityUid user, DoorComponent door, out DoAfterId? id, EntityUid tool)
+    {
+        id = null;
+
+        DoorPryingComponent? comp = null;
+        if (!Resolve(tool, ref comp))
+            return false;
+
+        if (comp.NeedsComponent)
+        {
+            ToolComponent? tComp = null;
+            if (!Resolve(tool, ref tComp) || !tComp.Qualities.Contains("Prying"))
+                return false;
+        }
+
+        if (!CanPry(target, user, comp.PryPowered))
+        {
+            return true;
+        }
+
+        StartPry(target, user, door, tool, comp.SpeedModifier, out id);
+        return true;
+    }
+
+    /// <summary>
+    ///     Pry open a door.
+    /// </summary>
+    public bool TryPry(EntityUid target, EntityUid user, DoorComponent door, out DoAfterId? id)
+    {
+        id = null;
+
+        if (!TryComp<PryUnpoweredComponent>(target, out _))
+            return false;
+
+        if (!CanPry(target, user, false))
+            return true;
+
+        StartPry(target, user, door, null, 1.0f, out id);
+        return true;
+    }
+
+    bool CanPry(EntityUid target, EntityUid user, bool pryPowered)
+    {
+        var canev = new BeforePryEvent(user, pryPowered);
+
+        RaiseLocalEvent(target, canev, false);
+
+        if (canev.Cancelled)
+            return false;
+        return true;
+    }
+
+    void StartPry(EntityUid target, EntityUid user, DoorComponent door, EntityUid? tool, float toolModifier, out DoAfterId? id)
+    {
+        var modEv = new DoorGetPryTimeModifierEvent(user);
+
+
+        RaiseLocalEvent(target, modEv, false);
+        var doAfterArgs = new DoAfterArgs(EntityManager, user, TimeSpan.FromSeconds(door.PryTime * modEv.PryTimeModifier / toolModifier), new DoorPryDoAfterEvent(), target, target, tool)
+        {
+            BreakOnDamage = true,
+            BreakOnUserMove = true,
+        };
+
+        if (tool != null)
+        {
+            _adminLog.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(user)} is using {ToPrettyString(tool.Value)} to pry {ToPrettyString(target)} while it is {door.State}");
+        }
+        else
+        {
+            _adminLog.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(user)} is prying {ToPrettyString(target)} while it is {door.State}");
+        }
+        _doAfterSystem.TryStartDoAfter(doAfterArgs, out id);
+    }
+
+    protected void OnDoAfter(EntityUid uid, DoorComponent door, DoorPryDoAfterEvent args)
+    {
+        if (args.Cancelled)
+            return;
+        if (args.Target is null)
+            return;
+
+        DoorPryingComponent? comp = null;
+
+        if (args.Used != null && Resolve(args.Used.Value, ref comp))
+            _audioSystem.PlayPredicted(comp.UseSound, args.Used.Value, args.User, comp.UseSound.Params.WithVariation(0.175f).AddVolume(-5f));
```

Comment:
Passing in manual audio params is deprecated and the sound should be specifying it.